lib/iex/lib/iex/info.ex

```diff
@@ -153,6 +154,20 @@ defimpl IEx.Info, for: List do
     [{"Description", description}, {"Reference modules", "Keyword, List"}]
   end
 
+  defp info_improper_list(_improper_list) do
+    description = """
+    This is what is referred to as an "improper list". An improper list is a
+    list which the last element in its internal representation is not a list.
+    For example: [1, 2, 3] is a proper list, as opposed to [1, 2 | 3] which is
+    an improper one.
+    """
```

Comment:
Feedback is welcome here, as there is a lot of room for improvement.

Reply:
I initially took that path, but then, I realized that `[1, 2 | [3]]` is a proper list

Reply:
@eksperimental that's why I included "last tail", because that list has another list so the last tail requires further traversing. We could make the definition recursive: "a proper list is a list where its tail is a non-empty proper list or the empty list" but I think that is less helpful. :)

Reply:
I have included your version